lightning/src/ln/peer_handler.rs

```diff
@@ -1272,4 +1284,124 @@ mod tests {
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_channel_update(&self, _msg: &msgs::ChannelUpdate) -> Result<bool, msgs::LightningError> {
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_htlc_fail_channel_update(&self, _update: &msgs::HTLCFailChannelUpdate) {}
+		fn get_next_channel_announcements(&self, starting_point: u64, batch_amount: u8) -> Vec<(msgs::ChannelAnnouncement, msgs::ChannelUpdate,msgs::ChannelUpdate)> {
+			let mut chan_anns = Vec::new();
+			let end: u64 = if starting_point > 0 { 6 } else { batch_amount.into() };
+			for j in starting_point..end {
+				let chan_upd_1 = get_dummy_channel_update(j);
+				let chan_upd_2 = get_dummy_channel_update(j);
+				let chan_ann = get_dummy_channel_announcement(j);
+
+				chan_anns.push((chan_ann, chan_upd_1, chan_upd_2));
+			}
+
+			chan_anns
+		}
+
+		fn get_next_node_announcements(&self, _starting_point: Option<&PublicKey>, _batch_amount: u8) -> Vec<msgs::NodeAnnouncement> {
+			Vec::new()
+		}
+
+		fn should_request_full_sync(&self, _node_id: &PublicKey) -> bool {
+			true
+		}
+	}
+
+	fn get_dummy_channel_announcement(short_chan_id: u64) -> msgs::ChannelAnnouncement {
+		let secp_ctx = Secp256k1::new();
+		let network = Network::Testnet;
+		let node_1_privkey = SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_2_privkey = SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_1_btckey = SecretKey::from_slice(&[40; 32]).unwrap();
+		let node_2_btckey = SecretKey::from_slice(&[39; 32]).unwrap();
+		let unsigned_ann = msgs::UnsignedChannelAnnouncement {
+			features: ChannelFeatures::supported(),
+			chain_hash: genesis_block(network).header.bitcoin_hash(),
+			short_channel_id: short_chan_id,
+			node_id_1: PublicKey::from_secret_key(&secp_ctx, &node_1_privkey),
+			node_id_2: PublicKey::from_secret_key(&secp_ctx, &node_2_privkey),
+			bitcoin_key_1: PublicKey::from_secret_key(&secp_ctx, &node_1_btckey),
+			bitcoin_key_2: PublicKey::from_secret_key(&secp_ctx, &node_2_btckey),
+			excess_data: Vec::new(),
+		};
+
+		let msghash = Message::from_slice(&Sha256dHash::hash(&unsigned_ann.encode()[..])[..]).unwrap();
+		let node_1_sig = secp_ctx.sign(&msghash, &node_1_privkey);
+		let node_2_sig = secp_ctx.sign(&msghash, &node_2_privkey);
+		let bitcoin_1_sig = secp_ctx.sign(&msghash, &node_1_btckey);
+		let bitcoin_2_sig = secp_ctx.sign(&msghash, &node_2_btckey);
+
+		msgs::ChannelAnnouncement {
+			node_signature_1: node_1_sig,
+			node_signature_2: node_2_sig,
+			bitcoin_signature_1: bitcoin_1_sig,
+			bitcoin_signature_2: bitcoin_2_sig,
+			contents: unsigned_ann,
+		}
+	}
+
+	fn get_dummy_channel_update(short_chan_id: u64) -> msgs::ChannelUpdate {
+		use secp256k1::ffi::Signature as FFISignature;
+		let network = Network::Testnet;
+		msgs::ChannelUpdate {
+			signature: Signature::from(FFISignature::new()),
```

Comment:
Why go to all the trouble to sign the channel_announcements only to use a dummy/invalid signature for channel_update?